src/colorizer/utils/texture_utils.ts

```diff
@@ -1,29 +1,52 @@
 import { DataTexture } from "three";
 
-import { FeatureDataType, featureTypeSpecs } from "../types";
+import { FeatureArrayType, FeatureDataType, featureTypeSpecs } from "../types";
 
 /**
- * Calculate the squarest possible texture that `data` can fit into, extend it with
- * `emptyVal` to fit perfectly into that space, and return a tuple of `[width, height]`
+ * Calculate the squarest possible texture that `data` can fit into and return a tuple of `[width, height]`
  */
-function fitIntoSquare<T>(data: T[], emptyVal: T): [number, number] {
+function getSquareDimensions<T extends FeatureDataType>(data: FeatureArrayType[T] | number[]): [number, number] {
```

Comment:
I would rename this to something like `getBestTextureDimensions`... calling it square when it will almost never actually be square can be misleading.